src/rules.ts

```diff
@@ -3,33 +3,46 @@ import {DefaultAzureCredential} from '@azure/identity';
 import {AzureClient} from './azure';
 import {ScanResult} from './scanner';
 
-export type Rule = IResourceGraphRule | IDummyRule;
+export type RuleContext = ResourceGraphRuleContext | DummyRuleContext;
 
-interface BaseRule {
+export interface ResourceGraphRuleContext {
+  type: 'resourceGraph';
+  rules: IResourceGraphRule[];
+  target: {
```

Comment:
We need to consider usage here. One subscription is fine. One resource group is fine. Multiple RG's could be okay. This design dictates that we can only scan multiple RG's in the same subscription. I'm less sure that is correct - it doesn't line up with how Resource Groups actually exist